config/FindVIGRANUMPY_DEPENDENCIES.cmake

```diff
@@ -2,65 +2,108 @@
+    execute_process(COMMAND ${PYTHON_EXECUTABLE} -c
+                     "import sys; print(sys.version_info[1])"
+                      OUTPUT_VARIABLE PYTHON_VERSION_MINOR OUTPUT_STRIP_TRAILING_WHITESPACE)
 
-    IF(APPLE AND ${PYTHON_PREFIX} MATCHES ".*framework.*")
-        SET(PYTHON_LIBRARIES "${PYTHON_PREFIX}/Python"
-            CACHE FILEPATH "Python libraries"
-            FORCE)
-    ELSE()
-        execute_process(COMMAND ${PYTHON_EXECUTABLE} -c
-                         "import sys; skip = 2 if sys.platform.startswith('win') else 1; print 'python' + sys.version[0:3:skip]"
-                          OUTPUT_VARIABLE PYTHON_LIBRARY_NAME OUTPUT_STRIP_TRAILING_WHITESPACE)
-        FIND_LIBRARY(PYTHON_LIBRARIES ${PYTHON_LIBRARY_NAME} HINTS "${PYTHON_PREFIX}" 
-                     PATH_SUFFIXES lib lib64 libs DOC "Python libraries")
-    ENDIF()
+    MESSAGE(STATUS "Using Python ${PYTHON_VERSION} at ${PYTHON_EXECUTABLE}")
 
-    # find Python includes
+    ######################################################################
+    #
+    #      find Python includes
+    #
+    ######################################################################
     execute_process(COMMAND ${PYTHON_EXECUTABLE} -c
-                    "from distutils.sysconfig import *; print get_python_inc()"
+                    "from distutils.sysconfig import *; print(get_python_inc())"
                     OUTPUT_VARIABLE PYTHON_INCLUDE OUTPUT_STRIP_TRAILING_WHITESPACE)
     SET(PYTHON_INCLUDE_PATH ${PYTHON_INCLUDE}
         CACHE PATH "Path to Python include files"
         FORCE)
 
-    IF(PYTHON_LIBRARIES AND PYTHON_INCLUDE_PATH)
-        MESSAGE(STATUS "Found Python libraries: ${PYTHON_LIBRARIES}")
+    IF(PYTHON_INCLUDE_PATH)
         MESSAGE(STATUS "Found Python includes:  ${PYTHON_INCLUDE_PATH}")
-        SET(PYTHONLIBS_FOUND TRUE)
     ELSE()
-        MESSAGE(STATUS "Could NOT find Python libraries and/or includes")
+        MESSAGE(STATUS "Could NOT find Python includes")
     ENDIF()
 
-    VIGRA_FIND_PACKAGE( Boost 1.40.0 COMPONENTS python )
+    ######################################################################
+    #
+    #      find Python library
+    #
+    ######################################################################
+    IF(APPLE AND ${PYTHON_PREFIX} MATCHES ".*framework.*")
+        SET(PYTHON_LIBRARIES "${PYTHON_PREFIX}/Python"
+            CACHE FILEPATH "Python libraries"
+            FORCE)
+    ELSE()
+        IF(WIN32)
+            set(PYTHON_LIBRARY_NAME python${PYTHON_VERSION_MAJOR}${PYTHON_VERSION_MINOR})
+        ELSE()
+            execute_process(COMMAND ${PYTHON_EXECUTABLE} -c
+                             "from distutils.sysconfig import *; print(get_config_var('LDLIBRARY'))"
+                              OUTPUT_VARIABLE PYTHON_LIBRARY_NAME OUTPUT_STRIP_TRAILING_WHITESPACE)
+            execute_process(COMMAND ${PYTHON_EXECUTABLE} -c
```

Comment:
Before this was run on Windows and Linux, which worked for me. However, now it appears this is just being run on Linux. I think this is what is causing AppVeyor to fail.